src/xcode/ENA/ENA/Source/Services/LocalStatistics/LocalStatisticsProvider.swift

```diff
@@ -0,0 +1,73 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+import OpenCombine
+
+class LocalStatisticsProvider: LocalStatisticsProviding {
+
+	// MARK: - Init
+
+	init(client: LocalStatisticsFetching, store: Store) {
+		self.client = client
+		self.store = store
+	}
+
+	// MARK: - Internal
+
+	func latestLocalStatistics(administrativeUnit: String, eTag: String? = nil) -> AnyPublisher<SAP_Internal_Stats_LocalStatistics, Error> {
+		let etag = store.localStatistics?.lastLocalStatisticsETag
+
+		guard let cached = store.localStatistics, !shouldFetch() else {
+			return fetchLocalStatistics(administrativeUnit: administrativeUnit, eTag: etag).eraseToAnyPublisher()
+		}
+		// return cached data; no error
+		return Just(cached.localStatistics)
+			.setFailureType(to: Error.self)
+			.eraseToAnyPublisher()
+	}
+
+	// MARK: - Private
+
+	private let client: LocalStatisticsFetching
+	private let store: LocalStatisticsCaching
+
+	private func fetchLocalStatistics(administrativeUnit: String, eTag: String? = nil) -> Future<SAP_Internal_Stats_LocalStatistics, Error> {
+		return Future { promise in
+			self.client.fetchLocalStatistics(administrativeUnit: administrativeUnit, eTag: eTag) { result in
+				switch result {
+				case .success(let response):
+					// cache
+					self.store.localStatistics = LocalStatisticsMetadata(with: response)
```

Comment:
The response of this API is usually very big as it has several States with tens of districts and we are only trying to fetch one district so it will be better to filter the response and only save requested District .
other with we have to filter the whole response every time we open the app

Reply:
Good point, let's discuss in our alignment call tomorrow.

Reply:
@30mar define 'big'? Caching the latest state for all districts seems tempting if we can reduce the number of follow up calls – given the server content (ETag) didn't change.